src/linesearches_compat.jl

```diff
@@ -0,0 +1,234 @@
+        vector_transport_method=vector_transport_method,
+    )
+end
+function LineSearchesStepsize(
+    ls,
+    is;
+    retraction_method=ExponentialRetraction(),
+    vector_transport_method=ParallelTransport(),
+)
+    return LineSearchesStepsize{
+        typeof(ls),typeof(is),typeof(retraction_method),typeof(vector_transport_method)
+    }(
+        ls, is, retraction_method, vector_transport_method
+    )
+end
+
+function (cs::StepsizeStorage{<:LineSearchesStepsize})(
+    mp::AbstractManoptProblem,
+    s::AbstractManoptSolverState,
+    i::Int,
+    η=-get_gradient(s);
+    fp=get_cost(mp, get_iterate(s)),
+    kwargs...,
+)
+    css = cs.stepsize
+    M = get_manifold(mp)
+    p = get_iterate(s)
+    X = get_gradient(s)
+    p_tmp = copy(M, p)
+    X_tmp = copy(M, p, η)
+    Y_tmp = copy(M, p, η)
+    f = get_cost_function(get_objective(mp))
+    dphi_0 = real(inner(M, p, X, η))
+    #println("dphi_0 = ", dphi_0)
+
+    # guess initial alpha
+    get_initial_alpha(M, css.initial_step, cs, p, η, fp, dphi_0)
+    α0 = cs.alpha
+
+    cs.last_f_p = fp
+
+    # perform actual linesearch
+
+    function ϕ(α)
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        return f(M, p_tmp)
+    end
+    function dϕ(α)
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        get_gradient!(mp, X_tmp, p_tmp)
+        vector_transport_to!(M, Y_tmp, p, η, p_tmp, css.vector_transport_method)
+        return real(inner(M, p_tmp, Y_tmp, Y_tmp))
+    end
+    function ϕdϕ(α)
+        # TODO: optimize?
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        get_gradient!(mp, X_tmp, p_tmp)
+        vector_transport_to!(M, Y_tmp, p, η, p_tmp, css.vector_transport_method)
+        phi = f(M, p_tmp)
+        dphi = real(inner(M, p_tmp, Y_tmp, Y_tmp))
+        return (phi, dphi)
+    end
+
+    try
+        α, fp = css.linesearch(ϕ, dϕ, ϕdϕ, α0, fp, dphi_0)
+        return α
+    catch ex
+        if isa(ex, LineSearches.LineSearchException)
+            # maybe indicate failure?
+            return zero(dphi_0)
+        else
+            rethrow(ex)
+        end
+    end
+end
+
+function show(io::IO, cs::LineSearchesStepsize)
+    return print(
+        io,
+        "LineSearchesStepsize($(cs.linesearch); retraction_method=$(cs.retraction_method), vector_transport_method=$(cs.vector_transport_method))",
+    )
+end
+
+"""
+    nanmax(x, y)
+
+Maximum of arguments, ignoring those that are NaN.
+"""
+function nanmax(x, y)
+    isnan(x) && return y
+    isnan(y) && return x
+    return max(x, y)
+end
+function nanmax(x, y, z)
+    return nanmax(nanmax(x, y), z)
+end
+
+"""
+    nanmin(x, y)
+
+Minimum of arguments, ignoring those that are NaN.
+"""
+function nanmin(x, y)
+    isnan(x) && return y
+    isnan(y) && return x
+    return min(x, y)
+end
+
+# code below was adapted from https://github.com/JuliaNLSolvers/LineSearches.jl/blob/master/src/initialguess.jl
+
+function get_initial_alpha(
```

Comment:
this is basically an internal `get_initial_stepsize`? Or the `initial_guess`from my Armijo implementation? I am not sure.

Reply:
This is specific to LineSearches.jl, though maybe after another rework it could also be used by other Manopt.jl line searches? I haven't checked. It is similar to `armijo_initial_guess`  but independent of a particular line search method.

Reply:
ah ok, so in the long run it might. because yes it is similar to the initial guess one – but Nonmonontone&Wolfe (in Moat) do not have that (yet) for the others this does not make sense (e.g. the constant stepsize)